### PR TITLE
User crud

### PR DESCRIPTION
**Pull request by m-nikhil**
* Added user CRUD controller and services
* Used softdeletes and ensured in updates
* Added request validation using `ValidationPipe`
* Added `EntityNotFoundExceptionFilter`
* Placed common utils under `common` folder 
* Ensured proper class-validation and class-transformation
* Added class and function documentations
* Added openapi specifics to controllers
* Did fomat and lint
* Marked TODOs

### CODE REVIEW

File: src/modules/user/dto/request/createUser.dto.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  readonly firstName: string;
+
+  @Word()
+  readonly lastName: string;
+
+  @Email()
+  readonly email: string;
+
+  @Password()
+  readonly password: string;
```
Comment (Amirthananth) [inline src/modules/user/dto/request/createUser.dto.ts:22]: Shouldn't we have a Min/Max value for the password ?

File: src/modules/user/user.service.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    return this.userRepository.save(user).catch(e => {
+      throw new BadRequestException('User already exist');
+    });
+  }
+
+  /**
+   * get user by id or throw EntityNotFoundError
+   */
+  async getUserById(id: number): Promise<User> {
+    return this.userRepository.findOneOrFail(id);
```
Comment (Amirthananth) [inline src/modules/user/user.service.ts:41]: Looks like we don't have error handling here

File: src/modules/user/user.service.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+    this.userRepository.softDelete(id);
+    return id;
+  }
+
+  /**
+   * query user table
+   */
+  async queryUser(): Promise<User[]> {
+    return this.userRepository.find();
```
Comment (Amirthananth) [inline src/modules/user/user.service.ts:100]: Is this a @Todo ?

Comment (Amirthananth) [reply src/modules/user/user.service.ts:100]: Or are we trying to fetch the whole list of users?

Comment (m-nikhil) [reply src/modules/user/user.service.ts:100]: Its whole list of users, who are not soft-deleted

File: src/modules/user/user.service.ts (hunk 2 above)
Comment (m-nikhil) [reply src/modules/user/user.service.ts:41]: This fail throws 'EntityNotFoundExceptionFilter' error.

I am capturing the error using global filters registered in main.ts
Link: https://github.com/Startup-Exploration/AuthServer/pull/1/files/12b31b3bc7a805720257c12423affb45b0bdc017#diff-8cfead41d88ad47d44509a8ab0a109adR24

Comment (Amirthananth) [reply src/modules/user/user.service.ts:41]: Okay. Cool

File: src/modules/user/user.service.ts (hunk 3 above)
Comment (Amirthananth) [reply src/modules/user/user.service.ts:100]: Okay

**Pull request merged**